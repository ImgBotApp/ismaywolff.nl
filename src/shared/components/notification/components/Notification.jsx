/* eslint-disable default-case, consistent-return */

import styled from 'styled-components'
import { oneOf } from 'prop-types'

const setBackground = props => {
  switch (props.type) {
    case 'success':
      return 'var(--color-green-light)'
    case 'error':
      return 'var(--color-red-light)'
  }
}

const setColor = props => {
  switch (props.type) {
    case 'success':
      return 'var(--color-green-dark)'
    case 'error':
      return 'var(--color-red-dark)'
  }
}

const Notification = styled.div`
  background: ${setBackground};
  color: ${setColor};
  padding: var(--size-smaller);
  text-align: center;
`

Notification.propTypes = {
  type: oneOf(['error', 'success']).isRequired
}

export default Notification
