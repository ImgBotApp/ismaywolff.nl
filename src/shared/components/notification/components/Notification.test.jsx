import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Notification from './Notification'

describe('<Notification />', () => {
  it('renders error type correctly', () => {
    const wrapper = shallow(<Notification type="error" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders success type correctly', () => {
    const wrapper = shallow(<Notification type="success" />)
    expect(wrapper).toMatchSnapshot()
  })
})
