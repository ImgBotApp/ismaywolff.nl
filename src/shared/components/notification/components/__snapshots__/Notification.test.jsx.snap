// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> renders error type correctly 1`] = `
.c0 {
  background: var(--color-red-light);
  color: var(--color-red-dark);
  padding: var(--size-smaller);
  text-align: center;
}

<div
  className="c0"
  type="error"
/>
`;

exports[`<Notification /> renders success type correctly 1`] = `
.c0 {
  background: var(--color-green-light);
  color: var(--color-green-dark);
  padding: var(--size-smaller);
  text-align: center;
}

<div
  className="c0"
  type="success"
/>
`;
